src/main/java/uk/gov/ons/census/casesvc/service/EventService.java

```diff
@@ -35,22 +40,39 @@ public void processSampleReceivedMessage(
     int questionnaireType =
         QuestionnaireTypeHelper.calculateQuestionnaireType(
             caze.getCaseType(), caze.getRegion(), caze.getAddressLevel());
-    UacQidLink uacQidLink = uacService.buildUacQidLink(caze, questionnaireType);
-    uacService.saveAndEmitUacUpdatedEvent(uacQidLink);
-    caseService.saveCaseAndEmitCaseCreatedEvent(caze);
+    if (createCaseSample.isBulkProcessed()) {
+      caseService.saveCaseAndEmitCaseCreatedEvent(
+          caze, buildMetadata(EventTypeDTO.SAMPLE_LOADED, ActionInstructionType.CREATE));
 
-    eventLogger.logCaseEvent(
-        caze,
-        OffsetDateTime.now(),
-        CREATE_CASE_SAMPLE_RECEIVED,
-        EventType.SAMPLE_LOADED,
-        createEventDTO(EventTypeDTO.SAMPLE_LOADED),
-        convertObjectToJson(createCaseSample),
-        messageTimestamp);
-
-    if (QuestionnaireTypeHelper.isQuestionnaireWelsh(caze.getTreatmentCode())) {
-      uacQidLink = uacService.buildUacQidLink(caze, 3);
+      eventLogger.logCaseEvent(
+          caze,
+          OffsetDateTime.now(),
+          CREATE_BULK_CASE_SAMPLE_RECEIVED,
+          EventType.SAMPLE_LOADED,
+          createEventDTO(
+              EventTypeDTO.SAMPLE_LOADED,
+              BULK_PROCESSING_EVENT_CHANNEL,
+              BULK_PROCESSING_EVENT_SOURCE),
+          convertObjectToJson(createCaseSample),
+          messageTimestamp);
+    } else {
+      UacQidLink uacQidLink = uacService.buildUacQidLink(caze, questionnaireType);
       uacService.saveAndEmitUacUpdatedEvent(uacQidLink);
+      caseService.saveCaseAndEmitCaseCreatedEvent(caze);
+
+      eventLogger.logCaseEvent(
```

Comment:
feels like this code repetition was easily avoidable (i.e. just varying the event description isn't hard)